### ISSUE TITLE
images and files removed if the saved failed

### ISSUE THREAD

**Issue by omargawdat** [unfoldadmin/django-unfold#585]
**video for illustration:**
https://github.com/user-attachments/assets/c9dc2fe9-6bcb-4d61-bc21-d19436459f39

**Problem**: 
on creating a model that requires images or files it gets removed if the saved fails for any reason and I have to upload all of them again in each failure, it causes a problem as some models require many files to be uploaded so in failure all of that has to be uploaded again.

**Unfold version: 0.32
Django version: 5**

**Comment by vasylkyselytsia** [comment]
The same issue, it's maybe problem reason:
![image](https://github.com/user-attachments/assets/9c029e6b-0bc9-4e5e-8475-1372352acabf)

**Comment by vasylkyselytsia** [comment]
app.js 

```js
/*************************************************************
 * File upload path
 *************************************************************/
const fileInputUpdatePath = () => {
  Array.from(document.querySelectorAll("input[type=file]")).forEach((input) => {
    input.addEventListener("change", (e) => {
      const parts = e.target.value.split("\\");
      const placeholder =
        input.parentNode.parentNode.querySelector("input[type=text]");
      placeholder.setAttribute("value", parts[parts.length - 1]);
    });
  });
};
```

Now `input[type=file]` has more parents, so the best way to fix it find form-field container and find `input[type=text]` there.

```js
const fileInputUpdatePath = () => {
  Array.from(document.querySelectorAll("input[type=file]")).forEach((input) => {
    input.addEventListener("change", (e) => {
      const parts = e.target.value.split("\\");
      const placeholder = input.closest('.form-row').querySelector("input[type=text]");
      placeholder.setAttribute("value", parts[parts.length - 1]);
    });
  });
};
```

**Comment by lukasvinclav** [comment]
I think, you are talking about two different problems.

@vasylkyselytsia has problem when the file is uploaded the placeholder is not changing. This was already fixed in #716 

@omargawdat wants to have uploaded files always uploaded when the validation fails but this is default behaviour in Django admin so this is not a bug. Feel free to try it in default admin to see how it is supposed to work.
